core/src/main/java/org/infinispan/CacheStream.java

```diff
@@ -319,6 +321,59 @@
    <R1, A> R1 collect(Collector<? super R, A, R1> collector);
 
    /**
+    * Performs a <a href="package-summary.html#MutableReduction">mutable
+    * reduction</a> operation on the elements of this stream using a
+    * {@code Collector} that is lazily created from the {@code SerializableSupplier}
+    * provided.
+    *
+    * This method behaves exactly the same as {@link #collect(Collector)} with
+    * the enhanced capability of working even when the mutable reduction
+    * operation has to run in a remote node and the operation is not
+    * {@link Serializable} or otherwise marshallable.
+    *
+    * So, this method is specially designed for situations when the user
+    * wants to use a {@link Collector} instance that has been created by
+    * {@link java.util.stream.Collectors} static factory methods.
+    *
+    * In this particular case, the function that instantiates the
+    * {@link Collector} will be marshalled according to the
+    * {@link Serializable} rules.
+    *
+    * @param supplier The supplier to create the collector that is specifically serializable
+    * @param <R1> The resulting type of the collector
+    * @return the collected value
+    * @since 9.2
+    */
+   <R1> R1 collect(SerializableSupplier<Collector<? super R, ?, R1>> supplier);
```

Comment:
I find it a lot easier to have this method be a default method that just passes the `supplier` to the non serializable method variant. This way all implementations only have to implement the method once. And we don't really care how it is serialized in the impl, just that it is.